modules/apps/frontend-data-set/frontend-data-set-taglib/src/main/java/com/liferay/frontend/data/set/taglib/servlet/taglib/HeadlessDisplayTag.java

```diff
@@ -306,87 +299,74 @@ protected void cleanUp() {
-			"frontend-data-set:headless-display:fdsFiltersContext",
-			_fdsFiltersContext);
-		httpServletRequest.setAttribute(
-			"frontend-data-set:headless-display:fdsPaginationEntries",
-			_fdsPaginationEntries);
-		httpServletRequest.setAttribute(
-			"frontend-data-set:headless-display:fdsSortItemList",
-			_fdsSortItemList);
-		httpServletRequest.setAttribute(
-			"frontend-data-set:headless-display:formId", _formId);
-		httpServletRequest.setAttribute(
-			"frontend-data-set:headless-display:formName", _formName);
-		httpServletRequest.setAttribute(
-			"frontend-data-set:headless-display:id", _id);
-		httpServletRequest.setAttribute(
-			"frontend-data-set:headless-display:itemsPerPage", _itemsPerPage);
-		httpServletRequest.setAttribute(
-			"frontend-data-set:headless-display:namespace", _namespace);
-		httpServletRequest.setAttribute(
-			"frontend-data-set:headless-display:nestedItemsKey",
-			_nestedItemsKey);
-		httpServletRequest.setAttribute(
-			"frontend-data-set:headless-display:nestedItemsReferenceKey",
-			_nestedItemsReferenceKey);
-		httpServletRequest.setAttribute(
-			"frontend-data-set:headless-display:pageNumber", _pageNumber);
-		httpServletRequest.setAttribute(
-			"frontend-data-set:headless-display:paginationSelectedEntry",
-			_paginationSelectedEntry);
-		httpServletRequest.setAttribute(
-			"frontend-data-set:headless-display:portletURL", _portletURL);
-		httpServletRequest.setAttribute(
-			"frontend-data-set:headless-display:selectedItems", _selectedItems);
-		httpServletRequest.setAttribute(
-			"frontend-data-set:headless-display:selectedItemsKey",
-			_selectedItemsKey);
-		httpServletRequest.setAttribute(
-			"frontend-data-set:headless-display:selectionType", _selectionType);
-		httpServletRequest.setAttribute(
-			"frontend-data-set:headless-display:showManagementBar",
-			_showManagementBar);
-		httpServletRequest.setAttribute(
-			"frontend-data-set:headless-display:showPagination",
-			_showPagination);
-		httpServletRequest.setAttribute(
-			"frontend-data-set:headless-display:showSearch", _showSearch);
-		httpServletRequest.setAttribute(
-			"frontend-data-set:headless-display:style", _style);
+		if (Validator.isNotNull(_formId)) {
+			props.put("formId", _formId);
+		}
+
+		props.put("id", getId());
+		props.put("itemsActions", getFdsActionDropdownItems());
+		props.put("namespace", getNamespace());
+
+		if (Validator.isNotNull(_nestedItemsKey)) {
+			props.put("nestedItemsKey", _nestedItemsKey);
+		}
+
+		if (Validator.isNotNull(_nestedItemsReferenceKey)) {
+			props.put("nestedItemsReferenceKey", _nestedItemsReferenceKey);
+		}
+
+		props.put(
+			"pagination",
+			HashMapBuilder.<String, Object>put(
+				"deltas", _fdsPaginationEntries
+			).put(
+				"initialDelta", getItemsPerPage()
+			).put(
+				"initialPageNumber", getPageNumber()
+			).build());
+
+		props.put("portletId", PortalUtil.getPortletId(getRequest()));
+
+		System.out.println(PortalUtil.getPortletId(getRequest()));
```

Comment:
This is probably something we forgot to remove

Reply:
Well, that's embarrassing :joy:

Reply:
No problem, this can happen to all of us :smile: Happy new year @markocikos !

Reply:
Thank you, to you as well! :hugs: